sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/HttpTransportClient.java

```diff
@@ -361,6 +361,9 @@ private HttpRequest prepareHttpMessage(
                 throw new IllegalStateException();
         }
 
+        // add operation type
+        httpRequestMessage.withOperationType(request.getOperationType());
```

Comment:
HtpReqeustMessage CTOR always forces the operationType, why is it missing?
Ideally its better to make it operationType immutable.

Reply:
@kirankumarkolli - `HttpRequestMessage` constructor does not enforce `operationType` always, there are few constructors where we don't enforce it, however, we can make it immutable by some small code changes. 
@moderakh  also pointed out another approach, where we don't even need the `operationType` on the `HttpRequestMessage`, I think that will be more cleaner, will implement that approach.

Reply:
This is done, resolving the conversation.